### ISSUE TITLE
Tinkers Exploit/Possible Solution

### ISSUE THREAD

**Issue by NihilEntropy** [Melonslise/Locks#18]
I can't vouch for versions after 1.12.2, but the tinkers crafting table can still access a locked chest. So users in my mod pack using only 1 wood log can unlock any chest.

Looking into another issue, it would seem the chests are still accessible via other means too. I am unfamiliar with minecraft modding but I would assume you must prevent chests from being interacted with or recognized. As you said in the other post, you are unsure how to fix this. Could somehow moving the NBT data from the chest to the lock make it where ALL mods and such think the chest is "empty" until it is unlocked? Perhaps a "hopper" lock to disable this feature, for just locking chests in machines?

This solution however has the risk of losing items if something goes wrong, so some-kind of protection should be in place beforehand.

**Comment by MercurialPony** [comment]
I don’t understand anything. How do you bypass the locks? What other means are you talking about? Explain clearly please

**Comment by NihilEntropy** [comment]
In the tinkers mod, there is a crafting station that has a gui showing the inventory next to it, which unfortunately includes locked chests. The other means I was referring to hoppers, etc.

I saw tinkers change log that said something about this, they might have fixed it?

**Comment by MercurialPony** [comment]
I see. This isn’t strictly a bug, because I haven’t actually done anything to restrict inventory access, but this is planned

**Comment by Dimentive** [comment]
I’m assuming that locked inventories such as chests can also be accessed by conduits, item interfaces, and other blocks that can interact with inventories from other mods.

I don’t know how difficult this would be to implement, but it would be nice if a locked chest or another locked tile entity with an inventory is impossible to interact with by _anything_. Perhaps it could fool other mods by pretending not to have an inventory when it’s locked? I’m not even sure if this is possible, but if it is, it could be something you could look into.

**Comment by MercurialPony** [comment]
I tried looking into restricting inventory access. I'm afraid I won't add anything like this anytime soon. That would require some very invasive and complex coremod hacks, which I cannot pull off at the moment

**Comment by Dimentive** [comment]
Ah, that’s a shame.
